src/Microsoft.Data/DataFrame.IO.cs

```diff
@@ -0,0 +1,261 @@
+                    case Type decimalType when decimalType == typeof(decimal):
+                        BaseColumn decimalColumn = new PrimitiveColumn<decimal>(names[i]);
+                        df.InsertColumn(i, decimalColumn);
+                        break;
+                    case Type doubleType when doubleType == typeof(double):
+                        BaseColumn doubleColumn = new PrimitiveColumn<double>(names[i]);
+                        df.InsertColumn(i, doubleColumn);
+                        break;
+                    case Type floatType when floatType == typeof(float):
+                        BaseColumn floatColumn = new PrimitiveColumn<float>(names[i]);
+                        df.InsertColumn(i, floatColumn);
+                        break;
+                    case Type intType when intType == typeof(int):
+                        BaseColumn intColumn = new PrimitiveColumn<int>(names[i]);
+                        df.InsertColumn(i, intColumn);
+                        break;
+                    case Type longType when longType == typeof(long):
+                        BaseColumn longColumn = new PrimitiveColumn<long>(names[i]);
+                        df.InsertColumn(i, longColumn);
+                        break;
+                    case Type sbyteType when sbyteType == typeof(sbyte):
+                        BaseColumn sbyteColumn = new PrimitiveColumn<sbyte>(names[i]);
+                        df.InsertColumn(i, sbyteColumn);
+                        break;
+                    case Type shortType when shortType == typeof(short):
+                        BaseColumn shortColumn = new PrimitiveColumn<short>(names[i]);
+                        df.InsertColumn(i, shortColumn);
+                        break;
+                    case Type uintType when uintType == typeof(uint):
+                        BaseColumn uintColumn = new PrimitiveColumn<uint>(names[i]);
+                        df.InsertColumn(i, uintColumn);
+                        break;
+                    case Type ulongType when ulongType == typeof(ulong):
+                        BaseColumn ulongColumn = new PrimitiveColumn<ulong>(names[i]);
+                        df.InsertColumn(i, ulongColumn);
+                        break;
+                    case Type ushortType when ushortType == typeof(ushort):
+                        BaseColumn ushortColumn = new PrimitiveColumn<ushort>(names[i]);
+                        df.InsertColumn(i, ushortColumn);
+                        break;
+                    case Type stringType when stringType == typeof(string):
+                        BaseColumn stringColumn = new StringColumn(names[i], 0);
+                        df.InsertColumn(i, stringColumn);
+                        break;
+                    default:
+                        throw new NotSupportedException();
+                }
+            }
+
+            // Fills values.
+            using (var st = createStream())
```

Comment:
We already have ALL the values in memory, right? Why read them from the stream again?

Reply:
No, we don't. `lines` only has a length of `guess_rows` (defaults to 10 rows)